site/content/en/docs/tasks/integrate_a_custom_job.md

```diff
@@ -0,0 +1,64 @@
+---
+title: "Integrate a custom Job"
+date: 2023-07-25
+weight: 8
+description: >
+  Integrate a custom Job with Kueue.
+---
+# Integrate a custom Job with Kueue
+
+Kueue integrates with a couple of jobs, including Kubernetes batch Job, MPIJob, RayJob or JobSet.  
+If there is another Job, which you would like supported, please consider contributing. 
+This page shows how to integrate a custom Job with Kueue.
+
+## Pre-requisites
+
+1. The custom Job CRD should have a suspend-like field. You can check this [page](https://kubernetes.io/docs/concepts/workloads/controllers/job/#suspending-a-job) for suspending a job.
+
+## What you need to specify
+
+1. Configuration
+2. Controller
+3. Webhook
+4. Adjust build system
+
+## Configuration
+
+Add your framework name to `.integrations.frameworks` in [controller_manager_config.yaml](https://github.com/kubernetes-sigs/kueue/blob/main/config/components/manager/controller_manager_config.yaml)
+
+## Controller
+
+Add a new folder in `./pkg/controller/jobs/` to host the implementation of the integration.
+
+1. Register the new framework in `func Init()`
+2. Add RBAC Authorization using [kubebuilder marker comments](https://book.kubebuilder.io/reference/markers/rbac.html)
+    - Add RBAC Authorization for priorityclasses, events, workloads, resourceflavors and any resources you want.
+3. Implement the `GenericJob` interface, and other optional [interfaces defined by the framework](https://github.com/kubernetes-sigs/kueue/blob/main/pkg/controller/jobframework/interface.go).
+
+## Webhook
+
+Create a webhook file in the dedicated directory for the integration.
+
+You can learn how to create webhook in this [page](https://book.kubebuilder.io/cronjob-tutorial/webhook-implementation.html).
+
+The framework provides some validation functions for common labels and annotations.
+
+Your webhook should have ability to suspend jobs.
+
+
+## Adjust build system
+Integrations can also live in an external repo, while still using the same integration framework.
```

Comment:
+1, I think it is ok to be mention, but be brief about integrations in external repos.

Later when, we have examples of integrations in external repos we may expand the section with examples and a guide.

Reply:
This looks like the wrong place to say this. See my comment in line 10.